src/request/change-password/ChangePassword.js

```diff
@@ -0,0 +1,211 @@
+     * @param {ChangePassword.resolve} resolve
+     * @param {reject} reject
+     */
+    constructor(request, resolve, reject) {
+        this._resolve = resolve;
+        this._request = request;
+        this._reject = reject;
+
+        /** @type {Key} */
+        // eslint-disable-next-line no-unused-expressions
+        this._key;
+
+        // Pages
+        /** @type {HTMLFormElement} */
+        this.$enterPassword = (document.getElementById(ChangePassword.Pages.ENTER_PASSWORD));
+        /** @type {HTMLFormElement} */
+        this.$setPassword = (document.getElementById(ChangePassword.Pages.SET_PASSWORD));
+        /** @type {HTMLFormElement} */
+        const $downloadFile = (document.getElementById(ChangePassword.Pages.DOWNLOAD_FILE));
+
+        // Elements
+        /** @type {HTMLFormElement} */
+        const $passwordGetter = (this.$enterPassword.querySelector('.password-box'));
+        /** @type {HTMLFormElement} */
+        const $passwordSetter = (this.$setPassword.querySelector('.password-setter-box'));
+        /** @type {HTMLDivElement} */
+        const $loginFileIcon = (this.$setPassword.querySelector('.login-file-icon'));
+        /** @type {HTMLAnchorElement} */
+        const $downloadLoginFile = ($downloadFile.querySelector('.download-login-file'));
+
+        // Components
+        this._passwordSetter = new PassphraseSetterBox($passwordSetter, { hideSkip: true });
+        this._loginFileIcon = new LoginFileIcon($loginFileIcon);
+        const downloadLoginFile = new DownloadLoginFile($downloadLoginFile);
+
+        this._passwordGetter = new PassphraseBox($passwordGetter, {
+            buttonI18nTag: 'passphrasebox-continue',
+            minLength: this._request.keyInfo.hasPin ? 6 : undefined,
+            hideCancel: true,
+            hideInput: !this._request.keyInfo.encrypted,
+        });
+
+        // Adapt to type of secret
+        if (this._request.keyInfo.type === Nimiq.Secret.Type.PRIVATE_KEY) {
+            this._loginFileIcon.setFileUnavailable(true);
+            /** @type {HTMLDivElement} */ (this.$enterPassword.querySelector('.nq-text')).classList.add('display-none');
+            /** @type {HTMLDivElement} */ (this.$setPassword.querySelector('.nq-text')).classList.add('hidden');
+        }
+
+        // Progress Indicators
+        // eslint-disable-next-line no-new
+        new ProgressIndicator(
+            document.querySelector(`#${ChangePassword.Pages.ENTER_PASSWORD} .progress-indicator`),
+            4,
+            1,
+        );
+        this._progressIndicator = new ProgressIndicator(
+            document.querySelector(`#${ChangePassword.Pages.SET_PASSWORD} .progress-indicator`),
+            4,
+            2,
+        );
+        // eslint-disable-next-line no-new
+        new ProgressIndicator(
+            document.querySelector(`#${ChangePassword.Pages.DOWNLOAD_FILE} .progress-indicator`),
+            4,
+            4,
```

Comment:
Please add a comment why this jumps from 2 to 4 and leaves out step 3 (because step 3 is used below for the password repetition).